girder_volview/dicom.py

```diff
@@ -0,0 +1,130 @@
+import datetime
+
+import pydicom
+import pydicom.valuerep
+import pydicom.multival
+import pydicom.sequence
+
+from girder import events
+from girder.models.item import Item
+from girder.models.file import File
+
+
+def setupEventHandlers():
+    events.bind("model.file.save.after", "girder_volview", handleFileSave)
+
+
+def handleFileSave(event):
+    return addDicomTagsToItem(event.info)
+
+
+# Code modified from https://github.com/girder/girder/blob/master/plugins/dicom_viewer/girder_dicom_viewer/__init__.py
+def addDicomTagsToItem(file):
+    """
+    Add DICOM tags to Item metadata.
+    """
+    itemId = file["itemId"]
+    if not itemId:
+        return
+    dicomMetadata = _parseFile(file)
+    if dicomMetadata is None:
+        return  # not a dicom file
+    itemMeta = {"dicom": dicomMetadata}
+    item = Item().load(itemId, force=True)
+    Item().setMetadata(item, itemMeta)
+
+
+def _coerceValue(value):
+    # For binary data, see if it can be coerced further into utf8 data.  If
+    # not, mongo won't store it, so don't accept it here.
+    if isinstance(value, bytes):
+        if b"\x00" in value:
+            raise ValueError("Binary data with null")
+        try:
+            value.decode("utf-8")
```

Comment:
Did you mean to do `value = value.decode()`?  (The `"utf-8"` parameter is the default in any modern python).

Reply:
Hmm, looks like some of that old "ask for forgiveness" control flow.  Changed to
```
    try:
            value.decode()
            return bytes(value)
```